src/main/java/com/intuit/graphql/orchestrator/keydirective/KeyDirectiveValidator.java

```diff
@@ -0,0 +1,81 @@
+ */
+public class KeyDirectiveValidator {
+
+  public void validateKeyArguments(TypeDefinition typeDefinition,
+      List<Argument> argumentList) throws InvalidLocationForKeyDirective, EmptyFieldsArgumentKeyDirective, InvalidKeyDirectiveFieldReference, NoFieldsArgumentForKeyDirective, MultipleArgumentsForKeyDirective {
+
+    String containerName = typeDefinition.getName();
+
+    validateKeyDirectiveLocation(typeDefinition, containerName);
+    validateKeyArgumentSize(argumentList, containerName);
+
+    Optional<Argument> argument = argumentList.stream().findFirst();
+    if(argument.isPresent()) {
+      validateKeyArgumentName(argument.get(), containerName);
+
+      final List<String> keyFieldList = Arrays.asList(argument.get().getValueWithVariable().getStringValue().trim().split(" "));
+      validateKeyFieldsArgumentSize(keyFieldList, containerName);
+
+      final List<String> existingFieldDefinitions = getTypeFieldDefinitions(typeDefinition).stream().map(FieldDefinition::getName).collect(Collectors.toList());
+      validateKeyFieldReferences(existingFieldDefinitions, keyFieldList, containerName);
+    }
+  }
+
+  private List<FieldDefinition> getTypeFieldDefinitions(TypeDefinition typeDefinition) {
+    return (typeDefinition instanceof InterfaceTypeDefinition) ? ((InterfaceTypeDefinition) typeDefinition).getFieldDefinition() : ((ObjectTypeDefinition) typeDefinition).getFieldDefinition();
+  }
+
+  private void validateKeyArgumentSize(List<Argument> argumentList, String containerName) throws MultipleArgumentsForKeyDirective {
+    if(argumentList.size() > 1) {
+      throw new MultipleArgumentsForKeyDirective(containerName);
+    }
+  }
+
+  private void validateKeyDirectiveLocation(TypeDefinition typeDefinition, String containerName) throws InvalidLocationForKeyDirective {
+    if(!(typeDefinition.eContainer() instanceof TypeSystemDefinition || typeDefinition.eContainer() instanceof InterfaceTypeDefinition)) {
+      throw new InvalidLocationForKeyDirective(containerName, typeDefinition.eContainer().eClass().getInstanceClassName());
+    }
+  }
+
+  private void validateKeyArgumentName(Argument argument, String containerName) throws NoFieldsArgumentForKeyDirective {
+    if(!argument.getName().equals("fields")) {
+      throw new NoFieldsArgumentForKeyDirective(containerName);
+    }
+  }
+
+  private void validateKeyFieldsArgumentSize(List<String> keyFields, String containerName) throws EmptyFieldsArgumentKeyDirective {
+    if(keyFields.isEmpty() || keyFields.stream().anyMatch(field -> field.equals(""))) {
+      throw new EmptyFieldsArgumentKeyDirective(containerName);
+    }
+  }
+
+  private  void validateKeyFieldReferences(List<String> existingFieldDefinitions, List<String> keyFieldRefereces, String containerName) throws InvalidKeyDirectiveFieldReference {
+    for(String keyField : keyFieldRefereces) {
+      if(existingFieldDefinitions.stream().noneMatch(definedField -> definedField.equals(keyField))) {
```

Comment:
not sure if I missed something, can we do `!existingFieldDefinitions.contains(keyField)`